loadbalancer: Port forwarding mode to new control-plane

joamaki opened the pull request:
This ports the service forwarding mode annotation to the new control-plane (https://github.com/cilium/cilium/pull/35064).

To simplify the implementation and testing I'm first moving the related configuration options to `loadbalancer.Config`.

joamaki committed:
loadbalancer: Port forwarding mode to new control-plane

This ports the service forwarding mode annotation to the new control-plane
(https://github.com/cilium/cilium/pull/35064).

Signed-off-by: Jussi Maki <jussi@isovalent.com>

borkmann reviewed (approved):
awesome, so much less code compared to the old control plane 🚀

joamaki committed:
cilium: Move LoadBalancerMode* to loadbalancer.Config

Move the load-balancer mode related configuration options into
the loadbalancer config. The LoadBalancerDSRDispatch validation
is commented out, but will be fixed in next commit.

Signed-off-by: Jussi Maki <jussi@isovalent.com>

joamaki committed:
cilium: Move LoadBalancerDSRDispatch to loadbalancer config

This moves the LoadBalancerDSRDispatch to loadbalancer.Config.

Signed-off-by: Jussi Maki <jussi@isovalent.com>

ti-mo reviewed (approved):
Thanks!

thorn3r reviewed (approved):
excellent :ship:

tommyp1ckles reviewed (approved):
@cilium/metrics lgtm

joestringer commented on pkg/loadbalancer/config.go:
```diff
@@ -322,6 +388,43 @@ func NewConfig(log *slog.Logger, userConfig UserConfig, deprecatedConfig Depreca
+		return Config{}, fmt.Errorf("The value --%s=%s is not supported as default under annotation mode", LoadBalancerModeName, cfg.LBMode)
+	}
+
+	/* FIXME:
+
+	if cfg.NodePortMode == option.NodePortModeDSR &&
+		cfg.LoadBalancerDSRDispatch != option.DSRDispatchOption &&
+		cfg.LoadBalancerDSRDispatch != option.DSRDispatchIPIP &&
+		cfg.LoadBalancerDSRDispatch != option.DSRDispatchGeneve {
+		return fmt.Errorf("Invalid value for --%s: %s", option.LoadBalancerDSRDispatch, cfg.LoadBalancerDSRDispatch)
+	}
+
+	if cfg.NodePortMode == option.NodePortModeHybrid &&
+		cfg.LoadBalancerDSRDispatch != option.DSRDispatchOption &&
+		cfg.LoadBalancerDSRDispatch != option.DSRDispatchGeneve {
+		return fmt.Errorf("Invalid value for --%s: %s", option.LoadBalancerDSRDispatch, cfg.LoadBalancerDSRDispatch)
+	}
+
+	if cfg.LoadBalancerModeAnnotation &&
+		cfg.LoadBalancerDSRDispatch != option.DSRDispatchIPIP {
+		return fmt.Errorf("Invalid value for --%s: %s", option.LoadBalancerDSRDispatch, cfg.LoadBalancerDSRDispatch)
+	}*/
```
For reviewers of this PR, particularly @cilium/sig-lb which looks like it is the likely owner and also missed reviewed: Please make sure you put eyes on the code. We generally do not comment out code with FIXME and merge.
